spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1217,16 +1217,18 @@ def refresh_container(self):
 
     # ---- For tabs
     # -------------------------------------------------------------------------
-    def add_tab(self, widget, name, filename='', give_focus=True):
+    def add_tab(self, client, name, filename='', give_focus=True):
         """Add tab."""
-        self.clients.append(widget)
-        index = self.tabwidget.addTab(widget, name)
+        if not isinstance(client, ClientWidget):
```

Comment:
Not sure if this is necessary, but with this I can get rid of `get_clients`, which is confusing as you can access the clients both with `get_clients` and `clients`